src/lib/auth/authconf.rb

```diff
@@ -0,0 +1,1089 @@
+            end
+            smb_conf = ad_create_tmp_smb_conf(@ad_domain, ad_get_workgroup_name(@ad_domain))
+            output = ''
+            exitstatus = 0
+            ou_param = @ad_ou.to_s == '' ? '' : "createcomputer=#{@ad_ou}"
+
+            Open3.popen2("net -s #{smb_conf.path} ads join #{ou_param} -U #{@ad_user}"){ |stdin, stdout, control|
+                stdin.print(@ad_pass + "\n")
+                stdin.close
+                output = stdout.read
+                exitstatus = control.value
+            }
+
+            # Get rid of the first output line that says "Enter XXX password"
+            output = output.split("\n").drop(1).join("\n")
+            # Optionally back up and save new samba configuration
+            if @ad_overwrite_smb_conf
+                path_original_smb_conf = '/etc/samba/smb.conf'
+                if File.exist?(path_original_smb_conf)
+                    ::FileUtils.copy_file(path_original_smb_conf, "#{path_original_smb_conf}.bak.#{Time.now.strftime('%Y%m%d%I%M%S')}", true, false)
+                end
+                ::FileUtils.copy_file(smb_conf.path, path_original_smb_conf, true, false)
+            end
+            smb_conf.unlink
+            return [exitstatus == 0, output]
+        end
+
+        # Return the PDC host name of the given AD domain via DNS lookup. If it cannot be found, return an empty string.
+        def ad_find_pdc(ad_domain_name)
+            begin
+                return Resolv::DNS.new.getresource("_ldap._tcp.pdc._msdcs.#{ad_domain_name}", Resolv::DNS::Resource::IN::SRV).target.to_s
+            rescue Resolv::ResolvError
+                return ''
+            end
+            return ''
+        end
+
+        # Return the KDC host name of the given AD domain via DNS lookup. If it cannot be found, return an empty string.
+        def ad_find_kdc(ad_domain_name)
+            begin
+                return Resolv::DNS.new.getresource("_kerberos._tcp.dc._msdcs.#{ad_domain_name}", Resolv::DNS::Resource::IN::SRV).target.to_s
+            rescue Resolv::ResolvError
+                return ''
+            end
+            return ''
+        end
+
+        # Read all authentication configuration items: kerberos, LDAP, pam and auxiliary daemons, and SSSD.
+        def read_all
+            krb_read
+            ldap_read
+            aux_read
+            sssd_read
+        end
+
+        # Return list of package names that should be installed to satisfy all configuration requirements.
+        def calc_pkg_deps
+            pkgs = []
+            if @sssd_enabled || @sssd_pam || @sssd_nss.any?
+                pkgs += ['sssd']
+                # Only install the required provider packages. By convention, they are named 'sssd-*'.
+                domain_providers = ['ad', 'ldap', 'ipa', 'proxy', 'krb5']
+                @sssd_conf.each { |_sect_name, conf|
+                    id_provider = conf['id_provider']
+                    auth_provider = conf['auth_provider']
```

Comment:
`rake check:syntax` (part of `rake osc:build`) says that this is an unused variable.
Also, `rubocop --lint` emits only 5 warnings (as opposed to the 4860 warnings for the regular run) and this is one of them. Maybe it is worth it to fix the 5 warnings?